sdk/src/types/block/core/basic.rs

```diff
@@ -139,6 +139,21 @@ impl BasicBlock {
     pub fn burned_mana(&self) -> u64 {
         self.burned_mana
     }
+
+    /// Returns the work score of a [`BasicBlock`].
+    pub fn workscore(&self, workscore_structure: WorkScoreStructure) -> u32 {
```

Comment:
This should definitely be a trait, since it's an identical definition across all impls

Reply:
yeah, I expected this comment ... I didn't do it immediatedly because it doesn't come with actual benefits for the user (from our perspective it's nice to have this internal abstraction, but from the outside the user probably shouldn't implement it and has to import it to even get access to it)

Reply:
Added the trait.